internal/storage/postgres/tree/create.go

```diff
@@ -60,10 +78,6 @@ func (r *TreeRepository) createEntity(ctx context.Context, entity *entities.Tree
 	var sensorID *int32
 	if entity.Sensor != nil {
 		sensorID = &entity.Sensor.ID
```

Comment:
Please do not delete the `UnlinkSensorIDFromTrees` here! Please also write a successful and error test case for this.

Reply:
Hello Dorien, sorry for the mistake. I accidentally removed the function call while copy&past from the old branch.

I tried to test the function with the following test, but somehow the function doesn't work correctly.
Here is my test and the result:

```go
t.Run("should unlink the sensor id in other trees", func(t *testing.T) {
		// given
		suite.ResetDB(t)
		suite.InsertSeed(t, "internal/storage/postgres/seed/test/tree")
		r := NewTreeRepository(suite.Store, mappers)

		sqlSensor, sensorErr := suite.Store.GetSensorByID(context.Background(), 1)
		if sensorErr != nil {
			t.Fatal(sensorErr)
		}
		sensor := mappers.sMapper.FromSql(sqlSensor)

		existTree, existTreeErr := r.GetByID(context.Background(), 1)
		if existTreeErr != nil {
			t.Fatal(sensorErr)
		}
		assert.NotNil(t, existTree.Sensor.ID)
		assert.Equal(t, sensor.ID, existTree.Sensor.ID)

		// when
		got, err := r.Create(context.Background(),
			WithSpecies("Oak"),
			WithTreeNumber("T001"),
			WithPlantingYear(2023),
			WithLatitude(54.801539),
			WithLongitude(9.446741),
			WithDescription("A newly planted oak tree"),
			WithWateringStatus(entities.WateringStatusGood),
			WithSensor(sensor),
		)

		sensorByNewTree, errSensorByTree := r.GetSensorByTreeID(context.Background(), got.ID)

		// then
		assert.NoError(t, err)
		assert.NotNil(t, got)
		assert.NotZero(t, got.ID)
		assert.WithinDuration(t, got.CreatedAt, time.Now(), time.Second)
		assert.WithinDuration(t, got.UpdatedAt, time.Now(), time.Second)
		assert.NoError(t, errSensorByTree)
		assert.NotNil(t, sensorByNewTree)
		assert.Equal(t, sensor.ID, sensorByNewTree.ID)
		assert.Nil(t, existTree.Sensor.ID)
```
The result:

```
2024/11/18 22:14:16 ERROR An Error occurred in database operation error="treecluster not found"
    create_test.go:138:
                Error Trace:    ../green-ecolution-backend/internal/storage/postgres/tree/create_test.go:138
                Error:          Expected nil, but got: 1
                Test:           TestTreeRepository_Create/should_unlink_the_sensor_id_in_other_trees
```
I expect that the ID in the old tree should be removed, but it still exists!

Please check my test above and if necessary, fix the function so that we can test it properly. Let me know if my test might be incorrect. Thank you!

Reply:
Ignore the test and please create a new issue.